safenode/src/node/api.rs

```diff
@@ -200,12 +202,19 @@ impl Node {
         match query {
             Query::Register(query) => self.registers.read(&query, User::Anyone).await,
             Query::GetChunk(address) => {
-                let resp = self
-                    .chunks
-                    .get(&address)
+                match self
+                    .network
+                    .get_provided_data(RecordKey::new(address.name()))
```

Comment:
So, if I'm not wrong, Client sends a `GetChunk` and nodes fetch the data for it? 
If so, maybe we can directly call `network.swarm.behaviour_mut().kademlia.get_record(key);` from the client. This has the advantage of short-circuiting when the data is found. Might speed up the queries by a bit.

Maybe we can do this in the next PR?

Reply:
> we can directly call network.swarm.behaviour_mut().kademlia.get_record(key); from the client

yes, I think we shall let client do this directly

Reply:
pushed change that makes client do the `get_record` directly.

Reply:
we may still need them for the later on usage, say some data exchanged among nodes.
so, I think we can keep them